### PR TITLE
*: make FakeMaster runnable

### PR DESCRIPTION
**Pull request by liuzix**

### CODE REVIEW

File: lib/fake/fake_master.go [hunk 1]
```diff
@@ -59,14 +75,15 @@ func (m *Master) OnWorkerMessage(worker lib.WorkerHandle, topic p2p.Topic, messa
...
 
-func NewFakeMaster(ctx *dcontext.Context, masterID lib.MasterID) lib.Master {
+func NewFakeMaster(ctx *dcontext.Context, _workerID lib.WorkerID, masterID lib.MasterID, _config lib.WorkerConfig) *Master {
 	ret := &Master{}
 	deps := ctx.Dependencies
 	base := lib.NewBaseMaster(
+		ctx,
 		ret,
 		masterID,
 		deps.MessageHandlerManager,
```
Comment (nongfushanquan) [inline lib/fake/fake_master.go:89]: how to receive those parameters

**Review by nongfushanquan — APPROVED** [review]

File: executor/server.go [hunk 2]
```diff
@@ -255,18 +252,27 @@ func (s *Server) Run(ctx context.Context) error {
...
 	}
 
+	err = s.startTCPService(ctx, wg)
+	if err != nil {
+		return err
+	}
+
 	// connects to metastore and maintains a etcd session
 	wg.Go(func() error {
+		log.L().Info("discoveryKeepalive")
```
Comment (amyangfei) [inline executor/server.go:275]:
```suggestion
```

File: client/executor_client.go [hunk 3]
```diff
@@ -60,10 +61,16 @@ func newExecutorClient(addr string) (*executorClient, error) {
 	if test.GetGlobalTestFlag() {
 		return newExecutorClientForTest(addr)
 	}
-	conn, err := grpc.Dial(addr, grpc.WithInsecure(), grpc.WithConnectParams(grpc.ConnectParams{Backoff: backoff.DefaultConfig}))
+	conn, err := grpc.Dial(
+		addr,
+		grpc.WithInsecure(),
+		grpc.WithConnectParams(grpc.ConnectParams{Backoff: backoff.DefaultConfig}),
+		grpc.WithBlock(),
+		grpc.WithUnaryInterceptor(grpc_zap.UnaryClientInterceptor(log.L().Logger)))
```
Comment (amyangfei) [inline client/executor_client.go:69]: How does this configuration work

File: executor/server.go [hunk 4]
```diff
@@ -143,33 +142,35 @@ func (s *Server) ResumeBatchTasks(ctx context.Context, req *pb.PauseBatchTasksRe
 
 func (s *Server) DispatchTask(ctx context.Context, req *pb.DispatchTaskRequest) (*pb.DispatchTaskResponse, error) {
 	log.L().Info("dispatch task", zap.String("req", req.String()))
-	workerID := s.idAllocator.AllocID()
```
Comment (amyangfei) [inline executor/server.go:146]: Could `idAllocator` be removed from the server struct?

**Pull request merged**